lxd/events/events.go

```diff
@@ -102,6 +103,44 @@ func (s *Server) AddListener(projectName string, allProjects bool, connection *w
 	return listener, nil
 }
 
+// AddListenerStream creates and returns a new event listener.
+func (s *Server) AddListenerStream(projectName string, allProjects bool, connection net.Conn, messageTypes []string, excludeSources []EventSource, recvFunc EventHandler, excludeLocations []string) (*Listener, error) {
```

Comment:
These 2 new functions (`AddListenerStream`) seem like unnecessary duplication. Are we able to just change the type of the `connection` argument on  the existing `AddListener` function to `EventListenerConnection`, and then have the correct `NewXListenerConnection()` function called by the caller of `AddListener` to pass in the standardised connection interface?